src/cfg/mod.rs

```diff
@@ -458,6 +474,34 @@ fn parse_defcfg(expr: &[SExpr]) -> Result<HashMap<String, String>> {
     }
 }
 
+/// Parse custom keys from an expression starting with defcustomkeys. Statefully updates the `keys`
+/// module using the custom keys parsed.
+fn parse_defcustomkeys(expr: &[SExpr]) -> Result<()> {
+    let mut cfg = HashMap::default();
+    let mut exprs = check_first_expr(expr.iter(), "defcustomkeys")?;
+    // Read k-v pairs from the configuration
+    loop {
+        let key = match exprs.next() {
+            Some(k) => k
+                .atom()
+                .ok_or_else(|| anyhow!("defcustomkeys contains a list; no lists are allowed"))?,
+            None => {
+                replace_custom_str_oscode_mapping(&cfg);
+                return Ok(());
+            }
+        };
+        let osc = match exprs.next() {
+            Some(v) => v.atom()
+                .ok_or_else(|| anyhow!("defcustomkeys contains a list; no lists are allowed"))
+                .and_then(|osc| osc.parse::<u16>().map_err(|_| anyhow!("defcustomkeys unknown number {osc}")))
+                .and_then(|osc| OsCode::from_u16(osc).ok_or_else(|| anyhow!("defcustomkeys unknown number {osc}")))?,
```

Comment:
I'm pretty sure this is incorrect; this should use `as`, since `from_u16` will differ between Linux and Windows. Probably good to add a unit test for this - make sure mapping between Linux and Windows is the same